### PR TITLE
feat(33-search): Product suggestions on search

### PR DESCRIPTION
**Pull request by ncuillery**

### CODE REVIEW

File: lib/product_list/cubits/product_search_cubit.dart [hunk 1]
```diff
@@ -9,10 +9,26 @@ import 'package:get_it/get_it.dart';
...
         ));
 
+  Future<void> searchProductsAutocomplete({
+    required String query,
+  }) async {
+    emit(state.copyWith(status: Status.pending));
+
+    final searchResults = await GetIt.instance
+        .get<ProductListApi>()
+        .searchProducts(query: query, limit: 5);
```
Comment (stephenrichter) [inline lib/product_list/cubits/product_search_cubit.dart:23]: configurable in the future, I'm guessing?

File: lib/flex_ui/widgets/ecommerce/product_list_item.dart [hunk 2]
```diff
@@ -57,7 +58,10 @@ class ProductListItem extends StatelessWidget {
                         rating: product.averageRating!,
                         size: 20,
                       ),
-                    Text(product.price!.formattedValue),
+                    ProductPrice(
```
Comment (stephenrichter) [inline lib/flex_ui/widgets/ecommerce/product_list_item.dart:61]: We may need a null check for this one. I've seen some client implementations where price is not set on a product for specific use cases/reasons (i.e. login required to see pricing).

File: lib/search/search_page.dart [hunk 3]
```diff
@@ -32,9 +41,39 @@ class SearchView extends StatelessWidget {
...
           ),
         ),
       ),
+      body: BlocBuilder<ProductSearchCubit, ProductSearchState>(
+        builder: (_, state) {
+          final children = <Widget>[
+            if (state.status == Status.pending && state.products.isEmpty)
+              const Center(child: CircularProgressIndicator()),
+            for (var p in state.products) ...[
+              Container(
```
Comment (stephenrichter) [inline lib/search/search_page.dart:57]: `Divider(height: 1, color: Colors.black.withOpacity(0.1),),` 😉

File: lib/search/widgets/autocomplete_product.dart [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      child: Padding(
+        padding: padding,
+        child: Row(
+          children: [
+            CachedImage(
+              url:
+                  '${dotenv.get('HYBRIS_BASE_URL')}${product.images.firstOrNull?.url}',
+              fit: BoxFit.contain,
+              placeholderAspectRatio: 1,
+              height: 64,
```
Comment (stephenrichter) [inline lib/search/widgets/autocomplete_product.dart:41]: Hmm, can we turn this into a FlexSize? I've had a few times I'd like to reuse this smaller thumbnail.
`static const double imageThumbSizeSm = 64.0;`

File: lib/search/widgets/autocomplete_product.dart [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            ),
+            const SizedBox(width: FlexSizes.spacerItems),
+            Expanded(
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.stretch,
+                children: [
+                  RichText(
+                    maxLines: null, // Authorize multi-lines without limit
```
Comment (stephenrichter) [inline lib/search/widgets/autocomplete_product.dart:49]: Nice! 👍

**Review by stephenrichter — APPROVED** [review]

File: lib/search/search_page.dart (hunk 3 above)
Comment (ncuillery) [reply lib/search/search_page.dart:57]: oooooh that's better 👌

File: lib/flex_ui/widgets/ecommerce/product_list_item.dart (hunk 2 above)
Comment (ncuillery) [reply lib/flex_ui/widgets/ecommerce/product_list_item.dart:61]: Oh right

Comment (ncuillery) [reply lib/flex_ui/widgets/ecommerce/product_list_item.dart:61]: The products with variants too, they have a price range instead of a price

**Pull request merged**